### ISSUE TITLE
Implement SLSA Provenance v1 git-specific digests

### ISSUE THREAD

**Issue by zregvart** [tektoncd/chains#914]
### Feature request

The SLSA v1 specification [allows `"git*"`](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#gitcommit-gittree-gitblob-gittag) as algorithm value for `predicate.buildDefinition.resolvedDependencies.digest`. Currently [it is set](https://github.com/tektoncd/chains/blob/main/pkg/chains/formats/slsa/internal/material/material.go) to `"sha1"` and the value is the git commit id.

With `artifacts.pipelinerun.format=slsa/v2alpha2` or `artifacts.taskrun.format=slsa/v2alpha2`, the is set to:

```json
{
  "uri": "git+https://github.com/example/repository.git",
  "digest": {
    "sha1": "ed1523732400440e387440de86674af5fb159285"
  },
  "name": "inputs/result"
}
```

With git-specific digests this could be:

```json
{
  "uri": "git+https://github.com/example/repository.git",
  "digest": {
    "gitCommit": "7a1027da300b7653c731c9c9bfafa1566ed50d7d"
  },
  "name": "inputs/result"
}
```

### Use case

Makes the `resolvedDependencies`, in case of git sources, more useful. They can now point to commits, tags, blobs or branches. See https://github.com/in-toto/attestation/issues/145.

**Comment by lcarva** [comment]
@zregvart, to be clear, this is just about changing the attribute named `.digest.sha1` to `.digest.gitCommit` in this particular case? (The example from the description uses different digests, not sure if that's intentional.)

**Comment by zregvart** [comment]
> @zregvart, to be clear, this is just about changing the attribute named `.digest.sha1` to `.digest.gitCommit` in this particular case? (The example from the description uses different digests, not sure if that's intentional.)

No, the hash computation is [different](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#gitcommit-gittree-gitblob-gittag). This follows how the [git-hash-object](https://www.git-scm.com/docs/git-hash-object) is implemented.
